.circleci/config.yml

```diff
@@ -13,15 +13,16 @@ jobs:
       - run: docker login -u $DOCKER_USER -p $DOCKER_PASS
 
       - run:
-          name: Build and deploy the application image
+          name: Build and deploy the application Docker image
           command: |
             DOCKER_IMAGE=firefoxtesteng/dummy-app-02
             DOCKER_TAG=$CIRCLE_BRANCH
 
             if [ "${CIRCLE_BRANCH}" == "master" ]; then
               DOCKER_TAG="latest"
+              echo "Pushing to ${DOCKER_IMAGE}:${DOCKER_TAG}"
+              docker build -t $DOCKER_IMAGE:$DOCKER_TAG .
+              docker push $DOCKER_IMAGE:$DOCKER_TAG
```

Comment:
or possibly we could do an `else exit 0` here or something and move the `echo` and `docker [build|push]` commands back down a level. Then possibly control everything using some ENV variable. So if a repo _wants_ Docker tags for each PR, they just set `$DOCKER_TAG_PR` to `true`, otherwise the variable defaults to `false` and we only tag on merges. If that makes sense.

Or, perhaps I'm overthinking this.